packages/dappmanager/hostScripts/docker_update.sh

```diff
@@ -0,0 +1,343 @@
+#    -i | --install : installs docker engine using "package method". If error returns string error
+# compose
+#    -v | --version : returns string with docker compose version
+#    -i | --install : installs docker compose. If error returns string error
+# system: returns system info: OS, version and docker versions (compose and engine)
+
+#####################
+### CONFIGURATION ###
+#####################
+
+# non-case sensitive (only available in bash)
+shopt -s nocasematch
+
+# Exit on error
+set -e
+
+####################
+### ERRORS CODES ###
+####################
+
+# 2 => no match requirements
+# 3 => unable to retrieve architecture
+# 4 => unable to retrieve os info
+# 5 => docker does not exist
+# 6 => docker compose does not exist
+# 7 => docker engine is stable
+# 8 => docker compose is stable
+# 9 => architecture not supported
+# 10 => Debian version not supported by docker and/or dappnode
+# 12 => Only 1 argument accepted
+# 13 => Arg must be docker-engine or docker-compose
+# 14 => flag must be -i | --install OR -v | --version
+# 15 => Illegal to downgrade docker engine (current docker version > than stable docker version)
+# 16 => Illegal to downgrade docker compose (current compose version > than stable compose version)
+
+###################
+#####VARIABLES#####
+###################
+
+# Docker stable versions
+STABLE_DOCKER_ENGINE_VERSION="20.10.2" # Same for PKG and CLI
+STABLE_DOCKER_CONTAINERD_VERSION="1.4.3-1"
+
+# Docker compose stable versions
+STABLE_DOCKER_COMPOSE_VERSION="1.25.5"
+
+# Download
+WGET="wget -q -O"
+
+#####################
+##### FUNCTIONS #####
+#####################
+
+function check_requirements() {
+  # check is debian
+  if [[ "$ID" != "debian" ]]; then
+    echo "OS must be debian"
+    exit 2
+  fi
+
+  if [ "$INSTALL_OPTION" == "engine" ]; then
+    # get DOCKER_SERVER_VERSION and DOCKER_CLI_VERSION
+    get_docker_engine_version
+
+    # check if docker version is equal or lower than the stable version
+    if [[ $DOCKER_SERVER_VERSION == "$STABLE_DOCKER_ENGINE_VERSION" ]]; then
+      echo "docker engine version is already stable"
+      exit 7
+    elif $(dpkg --compare-versions ${STABLE_DOCKER_ENGINE_VERSION} "lt" ${DOCKER_SERVER_VERSION}); then 
+      echo "Illegal to downgrade docker engine"
+      exit 15
+    fi
+  else
+    # get DOCKER_SERVER_VERSION and DOCKER_CLI_VERSION
+    get_docker_compose_version
+
+    # check if docker compose version is equal or lower than the stable
+    if [[ "$DOCKER_COMPOSE_VERSION" == "$STABLE_DOCKER_COMPOSE_VERSION" ]]; then
+      echo "docker compose version is stable"
+      exit 8
+    elif $(dpkg --compare-versions ${STABLE_DOCKER_COMPOSE_VERSION} "lt" ${DOCKER_COMPOSE_VERSION}); then 
+      echo "Illegal to downgrade docker compose"
+      exit 16
+    fi
+  fi
+
+  # check if architecture is supported by docker
+  if [ "$ARCHITECTURE" !=  "amd64" ] && [ "$ARCHITECTURE" !=  "arm64" ] && [ "$ARCHITECTURE" !=  "armhf" ]; then
```

Comment:
Consider using this https://stackoverflow.com/questions/21157435/bash-string-compare-to-multiple-correct-values